src/core/control/Control.cpp

```diff
@@ -1442,6 +1461,27 @@ void Control::deletePage() {
     this->win->getXournal()->forceUpdatePagenumbers();
 }
 
+void Control::movePage(size_t oldPos, size_t newPos) {
+    Document* doc = getDocument();
+    if (oldPos == newPos || doc->getPageCount() <= 1 || oldPos > doc->getPageCount() || newPos > doc->getPageCount()) {
+        return;
+    }
```

Comment:
Same here (assert)
Also, the doc should probably be locked for getPageCount() as well.